common/functions/src/scalars/semi_structureds/array_get.rs

```diff
@@ -0,0 +1,135 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_datavalues::with_match_scalar_types_error;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::Function;
+use crate::scalars::FunctionContext;
+use crate::scalars::FunctionDescription;
+use crate::scalars::FunctionFeatures;
+
+#[derive(Clone)]
+pub struct ArrayGetFunction {
+    array_type: ArrayType,
+    display_name: String,
+}
+
+impl ArrayGetFunction {
+    pub fn try_create(display_name: &str, args: &[&DataTypeImpl]) -> Result<Box<dyn Function>> {
+        let data_type = args[0];
+        let path_type = args[1];
+
+        if !data_type.data_type_id().is_array() || !path_type.data_type_id().is_integer() {
+            return Err(ErrorCode::IllegalDataType(format!(
+                "Invalid argument types for function '{}': ({:?}, {:?})",
+                display_name.to_uppercase(),
+                data_type.data_type_id(),
+                path_type.data_type_id()
+            )));
+        }
+
+        let array_type: ArrayType = data_type.clone().try_into()?;
+        Ok(Box::new(ArrayGetFunction {
+            array_type,
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic().num_arguments(2))
+    }
+}
+
+impl Function for ArrayGetFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn return_type(&self) -> DataTypeImpl {
+        NullableType::new_impl(self.array_type.inner_type().clone())
+    }
+
+    fn eval(
+        &self,
+        _func_ctx: FunctionContext,
+        columns: &ColumnsWithField,
+        input_rows: usize,
+    ) -> Result<ColumnRef> {
+        let indexes = build_path_indexes(columns[1].column())?;
+
+        let array_column: &ArrayColumn = if columns[0].column().is_const() {
+            let const_column: &ConstColumn = Series::check_get(columns[0].column())?;
+            Series::check_get(const_column.inner())?
+        } else {
+            Series::check_get(columns[0].column())?
+        };
+
+        let inner_type = self.array_type.inner_type().data_type_id();
+        with_match_scalar_types_error!(inner_type.to_physical_type(), |$T| {
+            let inner_column: &<$T as Scalar>::ColumnType = Series::check_get(array_column.values())?;
+            let mut builder = NullableColumnBuilder::<$T>::with_capacity(input_rows);
+
+            for index in indexes.iter() {
```

Comment:
I think it's incorrect, what if index's size is not 1 ?

Reply:
if the index is from a table column, we will have multi indexes.
for example:
```sql
mysql> create table t (index uint64);
mysql> insert into t values(0),(1),(2);
mysql> select get([1,2,3,4], index) from t;
+--------------------------+
| get([1, 2, 3, 4], index) |
+--------------------------+
|                        1 |
|                        2 |
|                        3 |
+--------------------------+
3 rows in set (0.05 sec)
```